tests/test_api_hosts_update.py

```diff
@@ -48,46 +46,43 @@ def test_update_fields(patch_doc, event_producer_mock, db_create_host, db_get_ho
         assert getattr(record, key) == patch_doc[key]
 
 
-@pytest.mark.parametrize("checkin_frequency", [1, 60, 1440])
-@pytest.mark.system_culling
-def test_checkin(checkin_frequency, event_datetime_mock, event_producer_mock, db_create_host, db_get_host, api_put):
-    host = db_host()
-    created_host = db_create_host(host)
+def test_checkin(event_datetime_mock, event_producer_mock, db_create_host, db_get_host, api_post):
+    created_host = db_create_host()
 
-    put_doc = {
-        "canonical_facts": {"insights_id": f"{created_host.canonical_facts['insights_id']}"},
-        "checkin_frequency": checkin_frequency,
-    }
+    post_doc = created_host.canonical_facts
+    updated_time = created_host.modified_on
 
-    expected_stale_timestamp = now() + timedelta(minutes=checkin_frequency)
-    response_status, response_data = api_put(
-        build_host_checkin_url(), put_doc, extra_headers={"x-rh-insights-request-id": "123456"}
+    response_status, response_data = api_post(
+        build_host_checkin_url(), post_doc, extra_headers={"x-rh-insights-request-id": "123456"}
     )
 
     assert_response_status(response_status, expected_status=201)
     record = db_get_host(created_host.id)
 
-    assert (record.stale_timestamp > expected_stale_timestamp) and (
-        record.stale_timestamp < expected_stale_timestamp + timedelta(seconds=1)
-    )
+    assert record.modified_on > updated_time
 
-    assert event_producer_mock.key == str(host.id)
+    assert event_producer_mock.key == str(created_host.id)
     assert_patch_event_is_valid(
-        host, event_producer_mock, "123456", event_datetime_mock, host.display_name, record.stale_timestamp, "checkin"
+        created_host,
+        event_producer_mock,
+        "123456",
+        event_datetime_mock,
+        created_host.display_name,
+        record.stale_timestamp,
     )
 
 
 @pytest.mark.system_culling
-def test_checkin_no_matching_host(event_producer_mock, db_create_host, db_get_host, api_put):
-    host = db_host()
-    created_host = db_create_host(host)
+def test_checkin_no_matching_host(event_producer_mock, db_create_host, db_get_host, api_post):
+    created_host = db_create_host()
 
-    put_doc = {
-        "canonical_facts": {"insights_id": f"nomatch_{created_host.canonical_facts['insights_id']}"},
-        "checkin_frequency": 60,
-    }
+    post_doc = created_host.canonical_facts
+    post_doc["insights_id"] = f"nomatch_{created_host.canonical_facts['insights_id']}"
+
+    response_status, response_data = api_post(
+        build_host_checkin_url(), post_doc, extra_headers={"x-rh-insights-request-id": "123456"}
+    )
 
-    response_status, response_data = api_put(build_host_checkin_url(), put_doc)
     assert_response_status(response_status, expected_status=400)
```

Comment:
You correctly changed the response status code to _404 Not Found_ in this case. This test is now wrong and passes only because of the mistake above.

```suggestion
    assert_response_status(response_status, expected_status=404)
```